File: src/main/java/com/kingmang/lazurite/core/ClassDeclarations.java
package com.kingmang.lazurite.core;

import com.kingmang.lazurite.parser.AST.Statements.ClassDeclarationStatement;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

@NoArgsConstructor
public final class ClassDeclarations {

    @Getter
    private static final Map<String, ClassDeclarationStatement> declarations;
    static {
        declarations = new ConcurrentHashMap<>();
    }

    public static void clear() {
        declarations.clear();
    }
    
    public static ClassDeclarationStatement get(String key) {
        return declarations.get(key);
    }
    
    public static void set(String key, ClassDeclarationStatement classDef) {
        declarations.put(key, classDef);
    }
}

File: src/main/java/com/kingmang/lazurite/core/ClassDeclarations.kt
package com.kingmang.lazurite.core

import com.kingmang.lazurite.parser.ast.statements.ClassDeclarationStatement
import lombok.Getter
import lombok.NoArgsConstructor
import java.util.concurrent.ConcurrentHashMap

object ClassDeclarations {
    private val declarations: MutableMap<String, ClassDeclarationStatement> = HashMap()

    @Synchronized
    fun clear() =
        declarations.clear()

    @Synchronized
    operator fun set(key: String, classDef: ClassDeclarationStatement) {
        declarations[key] = classDef
    }

    @Synchronized
    operator fun get(key: String): ClassDeclarationStatement? =
        declarations[key]
}
